src/common/btree.c

```diff
@@ -1521,6 +1667,10 @@ dbtree_fetch(daos_handle_t toh, dbtree_probe_opc_t opc, uint32_t intent,
 		return -DER_NO_HDL;
 
 	rc = btr_probe_key(tcx, opc, intent, key);
+	if (rc == PROBE_RC_INPROGRESS) {
+		D_DEBUG(DB_TRACE, "Target is in some uncommitted DTX.\n");
+		return -DER_INPROGRESS;
+	}
```

Comment:
Is this EINPROGRESS to inform client to retry from leader instead of retry on same participant? If so, DER_NOTLEADER look like a better error code? Because EINPROGRESS is similar to EAGAIN, it's usually used to inform client to keep busy retry.

Reply:
When the client get -EINPROGRESS failure, it will retry the RPC with leader. Even if the current server is just the leader, it also may cause the client to retry. From the client view, it does not need to distinguish whether it is the leader as retry or non-leader ask to do that, it just uses the same logic: retry with leader.